python/django_bolt/testing/client.py

```
@@ -23,10 +25,65 @@
+    Requests run through the Rust pipeline, so handlers execute on framework
+    threads with their own database connections. A test that wraps itself in a
+    transaction — Django's ``TestCase``, or pytest-django's plain ``django_db``
+    — never commits, so those rows are invisible to the handler, and on SQLite
+    the write lock the test holds turns the handler's query into ``database
+    table is locked``. Either way the symptom is an opaque 500 with nothing
+    pointing at the cause.
+
+    Advisory only: a test whose handlers never touch the database works fine
+    inside a transaction, so this must not fail the run.
+    """
+    global _atomic_block_warning_emitted
```

Ruff (0.16.1)

[warning] Using the global statement to update `_atomic_block_warning_emitted` is discouraged

(PLW0603)